tools/dump_kubernetes.sh

```diff
@@ -258,18 +259,14 @@ check_logs_for_errors() {
 }
 
 main() {
+  local exit_code=0
   parse_args "$@"
   check_prerequisites kubectl
   dump_time
   dump_pilot
   dump_resources
-  tap_containers
-  dump_logs_for_container
-
-  local exit_code=0
-  if copy_core_dumps_if_istio_proxy; then
-    log "FAIL: core file detected! Please check prior message for location."
-    exit_code=255
+  if tap_containers dump_logs_for_container copy_core_dumps_if_istio_proxy; then
```

Comment:
I think you wanted a `!` here, otherwise if `exit_code` gets set to 254 when the command succeeds. However, you probably actually want:
`exit_code=tap_containers dump_logs_for_container copy_core_dumps_if_istio_proxy`

Then you could switch on `exit_code` if you want to log details like you had before: `log "FAIL: core file detected! Please check prior message for location.`